src/js/common/schemaform/helpers.js

```diff
@@ -309,7 +309,7 @@ export function getNonArraySchema(schema) {
  * If no required fields are changing, it makes sure to not mutate the existing schema,
  * so we can still take advantage of any shouldComponentUpdate optimizations
  */
-export function updateRequiredFields(schema, uiSchema, formData) {
+export function updateRequiredFields(schema, uiSchema, pageData, formData) {
```

Comment:
Methinks this naming convention makes more sense; `pageData` for the data on the current page, `formData` for the data in the whole form (whether broken up by page or not, though we should probably have a convention for that).